### ISSUE TITLE
BUG: [[wikilinks]] do not function w line labels as expected

### ISSUE THREAD

**Issue by faizoro** [zsviczian/obsidian-excalidraw-plugin#2023]
### Have you searched for existing issues (including closed ones)?

- [X] Yes, I have searched the existing issues.

### Does this bug persist in a new vault with only Excalidraw installed?

- [X] Yes, I have verified the issue persists.

### Your environment

SYSTEM INFO:
	Obsidian version: v1.7.1
	Installer version: v1.4.16
	Operating system: Windows 10 Pro 10.0.22631
	Login status: logged in
	Language: en
	Catalyst license: insider
	Insider build toggle: on
	Live preview: on
	Base theme: light
	Community theme: AnuPpuccin v1.5.0
	Snippets enabled: 1
	Restricted mode: off
	Plugins installed: 60
	Plugins enabled: 42
		1: Advanced Slides v1.20.0
		2: Advanced Tables v0.21.0
		3: Banners v1.3.3
		4: Book Search v0.7.4
		5: Buttons v0.5.1
		6: Commander v0.5.1
		7: Dataview v0.5.67
		8: DB Folder v3.5.1
		9: Emoji Toolbar v0.4.1
		10: ExcaliBrain v0.2.15
		11: Find orphaned files and broken links v1.10.1
		12: Homepage v4.0.4
		13: Hover External Link Plugin v1.0.0
		14: Link Favicons v1.8.4
		15: Markdown Table Editor v0.3.1
		16: Minimal Theme Settings v8.1.1
		17: Natural Language Dates v0.6.2
		18: Outliner v4.8.1
		19: Paste URL into selection v1.7.0
		20: PodNotes v2.13.2
		21: QuickAdd v1.11.1
		22: Recent Files v1.4.2
		23: Sentence Navigator v1.3.0
		24: Shortcuts extender v2.2.0
		25: Style Settings v1.0.9
		26: Tab Switcher v1.4.0
		27: Tag Wrangler v0.6.1
		28: Task Progress Bar v2.0.0
		29: Templater v2.7.1
		30: Workspaces Plus v0.3.3
		31: Zoom v1.1.2
		32: Leaflet v6.0.5
		33: Reminder v1.1.15
		34: Footnote Shortcut v0.1.3
		35: Better footnote v1.0.1
		36: Tidy Footnotes v0.1.1
		37: Excalidraw v2.4.1
		38: Note Toolbar v1.11.0
		39: Longform v2.0.7
		40: Periodic Notes v0.0.17
		41: Calendar v1.5.10
		42: Projects v1.17.4

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

### Describe the bug

So - this would almost be a 'feature request' but no - it's a bug. I'll explain shortly.
When I label a line I've created `[[Sigmund Freud]]` I would expect that the resulting label would behave as a link. That is, `ctrl` + `left click` should open the file (which does exist) in a new tab, or `ctrl` + `hover` should show a preview of the md file. It doesn't do this. This would be a feature request except for that I notice that there is a balloon or pin indicating the label was recognized and telagraphs it's status to the user as a link, though it does not behave as such: ![Clipboard_09-13-2024_01](https://github.com/user-attachments/assets/bf086810-298d-4ad4-b227-3fa1b9cb1420) 

### Steps to reproduce

1. in any drawing, create a line.
2. double click the line to create a label for the line.
3. use wiki-link to point to an existing MD file.
4. focus outside the edit window of the line, observe that it has the lable minus the verbose wikilink indication brackets, and plus the pin or link indicator (setting dependent).
5. try to use the link - it won't behave as expected by ctrl right-clicking or ctrl hovering. 

### Expected behavior

I would expect the label and or line to link to the valid MD file it points do.

**Comment by zsviczian** [comment]
This is indeed a bug. As a workaround you can add a link (CTRL+K) to the arrow. That will work.

Upon further testing I noticed that indeed if you click in the top right of the label, the link works... but still this is not as intended.

<img width="320" alt="image" src="https://github.com/user-attachments/assets/dd3d35ea-8d76-483f-aeef-e6edaf41efd1">

**Comment by zsviczian** [comment]
To manage expectations... I don't expect a quick solution here. I started to look into the issue, but this requires a full refactoring of the code. Over the years handling links has organically grown into a beast, so refactoring is timely... But, there are at least 40 different usa-cases to consider plus a complex list of different type of links all requiring different type of parsing. I tried to quick fix the arrow links, but because ctrl+click is used by Excalidraw for starting the linear element editor, there is no quick fix.

![image](https://github.com/user-attachments/assets/c4cd9bda-055e-4753-a97d-cc7bd36c647c)
